src/components/track-changes/TrackChangesPanel.tsx

```diff
@@ -16,72 +16,56 @@ import {
   trackCommands,
   TrackedChange,
 } from '@manuscripts/track-changes-plugin'
-import { NodeSelection, TextSelection } from 'prosemirror-state'
-import React, { useState } from 'react'
+import React, { useCallback, useEffect, useState } from 'react'
 
 import useExecCmd from '../../hooks/use-exec-cmd'
 import { useStore } from '../../store'
 import { SnapshotsDropdown } from '../inspector/SnapshotsDropdown'
 import { SortByDropdown } from './SortByDropdown'
 import { SuggestionList } from './suggestion-list/SuggestionList'
+import { setChangeStatus, setSelectedSuggestion } from './utils'
 
 export const TrackChangesPanel: React.FC = () => {
   const [sortBy, setSortBy] = useState('in Context')
 
-  const [{ editor, trackState, selectedSuggestionID }] = useStore((store) => ({
-    editor: store.editor,
-    trackState: store.trackState,
-    selectedSuggestionID: store.selectedSuggestionID,
-  }))
-
-  const setSelectedSuggestion = (suggestion: TrackedChange) => {
-    const state = editor.state
-    const view = editor.view
-    const tr = state.tr
-    if (suggestion.type === 'text-change') {
-      const pos = suggestion.to
-      tr.setSelection(TextSelection.create(state.doc, pos, pos))
-    } else {
-      tr.setSelection(NodeSelection.create(state.doc, suggestion.from))
-    }
+  const [{ trackState, selectedSuggestionID }, _, getState] = useStore(
+    (store) => ({
+      // @TODO - check if trackstate actually issues some updates and we listen to them
+      trackState: store.trackState,
+      selectedSuggestionID: store.selectedSuggestionID,
+    })
+  )
 
-    view?.focus()
-    view?.dispatch(tr.scrollIntoView())
-  }
+  const onSelect = useCallback((suggestion: TrackedChange) => {
```

Comment:
Can we call this `handleSelect` for consistency?

Reply:
done